shared/network/src/p2p_model_sharing.rs

```diff
@@ -91,123 +84,107 @@ impl PeerManagerHandle {
+    pub fn report_blob_ticket_request_error(
+        &self,
+        peer_id: NodeId,
+        blob_ticket: Option<BlobTicket>,
+    ) {
         if self
             .peer_tx
-            .send(PeerCommand::ReportBlobTicketRequestError { peer_id })
+            .send(PeerCommand::ReportModelDownloadError {
+                peer_id,
+                blob_ticket,
+            })
             .is_err()
         {
             tracing::error!("Failed to report error for peer {peer_id}, PeerManager actor is dead");
         }
     }
-
-    /// Report that a peer has failed to in the middle of the download process of a blob ticket
-    pub fn report_blob_ticket_download_error(&self, peer_id: NodeId) {
-        if self
-            .peer_tx
-            .send(PeerCommand::ReportBlobTicketDownloadError { peer_id })
-            .is_err()
-        {
-            tracing::error!(
-                "Failed to report retry error for peer {peer_id}, PeerManager actor is dead"
-            );
-        }
-    }
 }
 
 struct PeerManagerActor {
     /// Peers that are available to request the model to
     available_peers: VecDeque<NodeId>,
     /// A map for the peer's blob ticket to their errors
-    /// A node could success sending the blob ticket but fail in the middle of the download so we differentiate between the two
-    /// Node -> (blob_ticket_request_errors, blob_ticket_downloads_errors)
-    errors_per_peers: HashMap<NodeId, (u8, u8)>,
+    errors_per_peers: HashMap<NodeId, u8>,
     /// Max errors we tolerate for a peer to share a parameter blob ticket
     max_errors_per_peer: u8,
-    /// Max errors we tolerate for a peer to fail in the middle of the download of a parameter blob ticket
-    max_retries_per_peer: u8,
 }
 
 impl PeerManagerActor {
-    pub fn new(max_errors_per_peer: u8, max_retries_per_peer: u8) -> Self {
+    pub fn new(max_errors_per_peer: u8) -> Self {
         Self {
             available_peers: VecDeque::new(),
             errors_per_peers: HashMap::new(),
             max_errors_per_peer,
-            max_retries_per_peer,
         }
     }
 
     fn handle_message(&mut self, message: PeerCommand, cancellation_token: CancellationToken) {
         match message {
             PeerCommand::SetPeers { peers } => {
                 self.available_peers = VecDeque::from(peers);
-                let errors_per_peers_vec =
-                    self.available_peers.iter().map(|peer| (*peer, (0u8, 0u8)));
+                let errors_per_peers_vec = self.available_peers.iter().map(|peer| (*peer, 0_u8));
                 self.errors_per_peers = HashMap::from_iter(errors_per_peers_vec);
 
-                debug!(
+                info!(
                     "Updated peer list: {} peers available to ask for the model parameters",
                     self.available_peers.len()
                 );
             }
             PeerCommand::GetPeer { reply } => {
+                info!("AVAILABLE PEERS: {:?}", self.available_peers);
```

Comment:
```suggestion
                debug!("AVAILABLE PEERS: {:?}", self.available_peers);
```
I would leave this as debug since it's pretty verbose.

Or we could also figure out a way to shorten each pubkey for the clients, so for example show only the first 4 characters.

Reply:
It was just for testing purposes, I’m not sure it gives us much useful information. We already have some logs for when the available peers list is empty, and one for when we successfully get a peer assigned, so I think that should be enough.